### PR TITLE
[RFC] Inductor as Backend of Quantization in PyTorch 2.0 Export

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
RFC to add Inductor as Backend of Quantization in PyTorch 2.0 Export

### CODE REVIEW

File: rfc_x86_inductor_quantizer.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
+## Feature
+This RFC proposes to add [TorchInductor](https://dev-discuss.pytorch.org/t/torchinductor-a-pytorch-native-compiler-with-define-by-run-ir-and-symbolic-shapes/747) as one of the backends for [Quantization 2.0 in Export](https://pytorch.org/tutorials/prototype/quantization_in_pytorch_2_0_export_tutorial.html).
```
Comment (jgong5) [inline rfc_x86_inductor_quantizer.md:2]: I would suggest we narrow the scope of the RFC to x86 CPU while leave some notes as open discussions on how quantization of other devices can be supported with the inductor backend. For different devices, different quantizers have to be added while we should still be able to lower the quantized graph via inductor compiler. A question is how ARM cpu can be supported. We can probably still reuse the quantizer proposed here and leverage oneDNN/ARM as the backend.

File: rfc_x86_inductor_quantizer.md [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+* Previously, the weight of convolution is encapsulated in a `Conv2dPackedParamsBase` object instead of a tensor. Also, the qconv's implementation is encapsulated inside the `PackedConvWeight` class. Some runtime information for the calculation can be fetched from the attr of the `PackedConvWeight` object. For quantization 2.0, we will implement the new quantized convolution as functional invoking with only Tensor objects in inputs.
+
+Due to these differences in 1.X and 2.X path, we need to enable new quantized operators implementation.
+
+#### Weight prepack implementation
+OneDNN convolution needs prepacked weight to do the calculation. The weight prepack can be done in the graph preparation phase instead of the runtime to avoid runtime overhead. As we discuss [here](https://github.com/pytorch/pytorch/pull/101164#discussion_r1195953417), to do weight prepack we need the constant fold feature support in Inductor.
+
+* Current constant folding feature doesn't support FX graph captured by `torch._dynamo.export` as reported in [issue-103582](https://github.com/pytorch/pytorch/issues/103582).
```
Comment (jgong5) [inline rfc_x86_inductor_quantizer.md:117]: Is it an implementation issue or design issue?

Comment (leslie-fang-intel) [reply rfc_x86_inductor_quantizer.md:117]: I am not sure about it. May need further inputs from PyTorch team.

File: rfc_x86_inductor_quantizer.md (hunk 1 above)
Comment (leslie-fang-intel) [reply rfc_x86_inductor_quantizer.md:2]: Thanks and changed.

**Pull request closed without merging**